server/tests/test_manual_data_input.py

```diff
@@ -279,3 +273,61 @@ def test_modify_all_dimensions(vanilla_user_client):
         ).value
         == 7.5
     )
+
+    # check periods correctly assigned on all records
+    assert not ind.data.exclude(period=period)
+
+
+def test_non_changes_dont_create_versions(vanilla_user_client):
```

Comment:
Decided to move these to their own tests. Although we had something like this before (see deleted code) it actually wasn't testing stuff properly because it the "previous" records were not assigned the proper period